crypto/encode_decode/encoder_pkey.c

```diff
@@ -104,7 +104,10 @@ static void collect_encoder(OSSL_ENCODER *encoder, void *arg)
             || (encoder->does_selection != NULL
                 && !encoder->does_selection(provctx, data->ctx->selection)))
             continue;
-        (void)OSSL_ENCODER_CTX_add_encoder(data->ctx, encoder);
+
+        /* Only add each encoder implementation once */
+        if (OSSL_ENCODER_CTX_add_encoder(data->ctx, encoder))
```

Comment:
Should this error if this was to fail?
And then the break is a separate unconditional line.

Reply:
No.  If that fails, it simply means that this particular encoder implementation isn't added to the stack...  but what if there's another that works as well where this doesn't fail?  The end result is that the stack will only contain encoder implementations where this add succeeds.

Reply:
Failure may happen because the implementation didn't respond as expected to certain parameter queries.  That's ok.